CircleciScripts/setup_private_specs.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# This script generates a private spec repo in your $HOME/.aws-amplify
+# directory, to enable `pod lib lint`. It also makes it easier to do
+# development against unreleased versions of Amplify, at least until we start
+# releasing nightly builds.
+
+set -e
+
+declare -r LOCAL_SPEC_GIT_ROOT="$HOME/.aws-amplify/amplify-ios/amplify-ios-podspecs.git"
+declare -r LOCAL_SPEC_REPO_NAME="amplify-ios-local-specs"
+declare -r COCOAPODS_REPO_DIR="${HOME}/.cocoapods/repos/${LOCAL_SPEC_REPO_NAME}"
+
+function init_spec_git_root {
+  # If directory exists, assume it's correctly set up; don't attempt any
+  # repairs or validation
+  if [[ -d "${LOCAL_SPEC_GIT_ROOT}" ]] ; then
+    return 0
+  fi
+
+  mkdir -p "${LOCAL_SPEC_GIT_ROOT}"
+  old_pwd="${PWD}"
+  cd "${LOCAL_SPEC_GIT_ROOT}"
+  git init --bare
+  cd "${old_pwd}"
+}
+
+function init_spec_repo {
+  # If repo exists, assume it's correctly set up; don't attempt any
+  # repairs or validation
+  if [[ -n "$(pod repo list | grep ${LOCAL_SPEC_REPO_NAME})" ]] ; then
+    return 0
+  fi
+
+  init_spec_git_root
+
```

Comment:
if we return a value from init_spec_git_root, we can:
```
init_spec_git_root
retval=$?
if [[ $retval -ne 0 ]]; then
    return ${retval}
fi
```

Reply:
bash functions don't return values other than their exit status unfortunately :(

Reply:
Hmmm on line 18 (which is part of init_spec_get_root), we are doing a `return 0`... that being said.. i think you're right.. with regards to capturing the return value I think i usually set a global variable and check it.. in any case, we can handle this later.